ports/esp32/modnetwork.c

```diff
@@ -332,12 +337,57 @@ STATIC mp_obj_t esp_disconnect(mp_obj_t self_in) {
     return mp_const_none;
 }
 
+// Cases similar to ESP8266 user_interface.h
+// Error cases are referenced from wifi_err_reason_t in ESP-IDF
+enum {
+    STAT_IDLE                           = 1000,
+    STAT_CONNECTING                     = 1001,
+    STAT_GOT_IP                         = 1010
+};
+
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
 STATIC mp_obj_t esp_status(size_t n_args, const mp_obj_t *args) {
+    wlan_if_obj_t *self = MP_OBJ_TO_PTR(args[0]);
     if (n_args == 1) {
-        // no arguments: return None until link status is implemented
-        return mp_const_none;
+        if (self->if_id == WIFI_IF_STA) {
+            // Case of no arg is only for the STA interface
+            if (wifi_sta_connected) {
+                return MP_ROM_QSTR(MP_QSTR_STAT_GOT_IP);
+            } else if (wifi_sta_connect_requested) {
+                return MP_ROM_QSTR(MP_QSTR_STAT_CONNECTING);
+            } else {
+                switch (wifi_sta_disconn_reason) {
```

Comment:
And here you can just return `MP_OBJ_NEW_SMALL_INT(wifi_sta_disconn_reason)`.  Then you don't need any switch statement at all.  That was the reason to reuse the constants from the IDF.